spyder/app/tests/test_mainwindow.py

```diff
@@ -80,6 +80,41 @@ def close_widget():
 #==============================================================================
 @flaky(max_runs=10)
 @pytest.mark.skipif(os.name == 'nt', reason="It times out sometimes on Windows")
+def test_debug_error_code(main_window, qtbot):
+    """Test that the debug in IPython console handles errors in stdout."""
+    # Wait until the window is fully up
+    shell = main_window.ipyconsole.get_current_shellwidget()
+    control = shell._control
+    qtbot.waitUntil(lambda: shell._prompt_html is not None, timeout=SHELL_TIMEOUT)
+
+    # Load test file
+    test_file = osp.join(LOCATION, 'debug.py')
+    main_window.editor.load(test_file)
+
+    # Click the debug button
+    debug_action = main_window.debug_toolbar_actions[0]
+    debug_button = main_window.debug_toolbar.widgetForAction(debug_action)
+    qtbot.mouseClick(debug_button, Qt.LeftButton)
+    qtbot.wait(2000)
+    
+    # Run the current line
+    debug_next_action = main_window.debug_toolbar_actions[1]
+    debug_next_button = main_window.debug_toolbar.widgetForAction(debug_next_action)
+    qtbot.mouseClick(debug_next_button, Qt.LeftButton)
+    qtbot.wait(3000)
+    
+    # Run the current line
+    qtbot.mouseClick(debug_next_button, Qt.LeftButton)
+    qtbot.wait(3000)
+
+    assert "Kernel restarting because an error raised while debugging (stdout change)" in control.toPlainText()
```

Comment:
Please make this `assert` simpler, i.e. instead of verifying that this text is present in the console, just verify that there's no `a` value in the console (because the kernel was restarted :-).

For that, please modify the `get_value` method of `SpyderKernel` (in `utils/ipython/spyder_kernel.py`), like this

```python
def get_value(self, name):
   """Get the value of a variable"""
    ns = self._get_current_namespace()
    try:
        value = ns[name]
    except KeyError:
        value = (name, '__spy_missing_value__')
    publish_data({'__spy_data__': value})
```

and then change the assert line above like this

```python
assert shell.get_value('a') == ('a', '__spy_missing_value__')
```